core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -103,4 +104,16 @@ package object config {
     .stringConf
     .checkValues(Set("hive", "in-memory"))
     .createWithDefault("in-memory")
+
+  private[spark] val LISTENER_BUS_EVENT_QUEUE_SIZE =
+    ConfigBuilder("spark.scheduler.listenerbus.eventqueue.size")
+      .intConf
+      .transform((x: Int) => {
```

Comment:
The transform wasn't really meant to be a validator. We had discussed adding validator functionality to the Config stuff but I think that should be done separately.  So for now lets pull this out and just do the check in the LiveListenerBus.